dali/kernels/imgproc/paste/paste_gpu_input.h

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_IMGPROC_PASTE_PASTE_GPU_INPUT_H_
+#define DALI_KERNELS_IMGPROC_PASTE_PASTE_GPU_INPUT_H_
+
+namespace dali {
+namespace kernels {
+namespace paste {
+
+template<int ndims>
+struct GridCellInput {
+    int input_idx;
+    ivec<ndims> cell_start, cell_end, in_anchor;
+};
+
+template <int ndims>
+struct MultiPasteSampleInput {
+    int grid_cell_start_idx;
+    ivec<ndims> cell_counts;
+};
```

Comment:
Honestly, I have hard time deciphering what this "grid cell" is and what it's for.